src/main/java/de/unijena/cheminf/mortar/model/fragmentation/algorithm/AlkylStructureFragmenter.java

```diff
@@ -0,0 +1,1443 @@
+                for (IAtom tmpAtom: anAtomArray) {
+                    int tmpAtomIndex = tmpAtom.getProperty(AlkylStructureFragmenter.INTERNAL_ASF_ATOM_INDEX_PROPERTY_KEY);
+                    if ( tmpAtomIndex == tmpBeginIndex || tmpAtomIndex == tmpEndIndex) {
+                        tmpAtom.setProperty(AlkylStructureFragmenter.INTERNAL_ASF_TRIPLE_BOND_MARKER_KEY, true);
+                    }
+                }
+            }
+        }
+        aMolecularArraysInstance.setAtomArray(anAtomArray);
+        aMolecularArraysInstance.setBondArray(aBondArray);
+    }
+    /**
+     * Method to create a deeper copy of a given atom, meant to replace the default addAtom() method of IAtomContainer
+     * in which only the reference is added to the atomcontainer instead of a new atom.
+     *
+     * @param anAtomToCopy the atom to create a deep copy of
+     * @return the deep copy of the given atom
+     */
+    protected IAtom deepCopyAtom(IAtom anAtomToCopy) {
+        //seems like atom index is lost when deep copying is applied
+        IAtom tmpNewAtom = this.chemObjectBuilderInstance.newAtom();
+        tmpNewAtom.setAtomicNumber(anAtomToCopy.getAtomicNumber());
+        tmpNewAtom.setImplicitHydrogenCount(anAtomToCopy.getImplicitHydrogenCount());
+        tmpNewAtom.setCharge(anAtomToCopy.getCharge());
+        tmpNewAtom.setProperties(anAtomToCopy.getProperties());
+        return tmpNewAtom;
+    }
+    /**
+     * Method to create a deep copy of a given bond, meant to replace the default addBond() method of IAtomContainer
+     *
+     * @param aBondToCopy the bond to create a deep copy of
+     * @param aBondIncludingAtomContainer the atomcontainer in which the bond's atoms are placed in
+     * @return the deep copy of the given bond
+     */
+    protected IBond deepCopyBond(IBond aBondToCopy, IAtomContainer aBondIncludingAtomContainer) {
+        IBond tmpNewBond = this.chemObjectBuilderInstance.newBond();
+        tmpNewBond.setOrder(aBondToCopy.getOrder());
+        tmpNewBond.setProperties(aBondToCopy.getProperties());
+        IAtom tmpBeginAtom = aBondToCopy.getBegin();
+        int tmpBeginAtomIndex = tmpBeginAtom.getProperty(AlkylStructureFragmenter.INTERNAL_ASF_ATOM_INDEX_PROPERTY_KEY);
+        if ((tmpBeginAtom.getProperty(AlkylStructureFragmenter.INTERNAL_ASF_ATOM_INDEX_PROPERTY_KEY) != null)) {
+            int tmpEndAtomIndex = aBondToCopy.getEnd().getProperty(INTERNAL_ASF_ATOM_INDEX_PROPERTY_KEY);
+            for (IAtom tmpAtom : aBondIncludingAtomContainer.atoms()) {
+                if ((int) tmpAtom.getProperty(INTERNAL_ASF_ATOM_INDEX_PROPERTY_KEY) == tmpBeginAtomIndex) {
+                    tmpNewBond.setAtom(tmpAtom, 0);
+                } else if ((int) tmpAtom.getProperty(INTERNAL_ASF_ATOM_INDEX_PROPERTY_KEY) == tmpEndAtomIndex) {
+                    tmpNewBond.setAtom(tmpAtom, 1);
+                }
+            }
+        } else {
+            throw new IllegalArgumentException();
```

Comment:
must be declared in method head and doc and have a message describing its cause.